pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -854,6 +855,36 @@ protected BookieAffinityGroupData internalGetBookieAffinityGroup() {
         }
     }
 
+    private void validateLeaderBroker() {
+        if (!this.isLeaderBroker()) {
+            LeaderBroker leaderBroker = pulsar().getLeaderElectionService().getCurrentLeader().get();
+            String leaderBrokerUrl = leaderBroker.getServiceUrl();
+            try {
+                URL redirectUrl = new URL(leaderBrokerUrl);
+                URI redirect = UriBuilder.fromUri(uri.getRequestUri()).host(redirectUrl.getHost())
```

Comment:
it always returns http?

Reply:
shouldn't it be based on http-request that comes from user . I guess I gave code reference earlier?

Reply:
here uri is the input request uri. So if the input request has https, uri. getRequestUri() returns https url and we are only replacing host and port.